js/modules/k6/experimental/tracing/client.go

```diff
@@ -0,0 +1,243 @@
+package tracing
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	httpmodule "go.k6.io/k6/js/modules/k6/http"
+	"go.k6.io/k6/metrics"
+)
+
+// Client represents a HTTP Client instrumenting the requests
+// it performs with tracing information.
+type Client struct {
+	vu modules.VU
+
+	// opts holds the client's configuration options.
+	opts options
+
+	// propagator holds the client's trace propagator, used
+	// to produce trace context headers for each supported
+	// formats: w3c, b3, jaeger.
+	propagator Propagator
+
+	// requestFunc holds the http module's request function
+	// used to emit HTTP requests in k6 script. The client
+	// uses it under the hood to emit the requests it
+	// instruments.
+	requestFunc HTTPRequestFunc
+}
+
+type (
+	// HTTPRequestFunc is a type alias representing the prototype of
+	// k6's http module's request function
+	HTTPRequestFunc func(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error)
+)
+
+// NewClient instantiates a new tracing Client
+func NewClient(vu modules.VU, opts options) *Client {
+	rt := vu.Runtime()
+
+	// Instantiate the http module in our runtime and get its default exported object
+	httpModuleObj, ok := httpmodule.New().NewModuleInstance(vu).Exports().Default.(*goja.Object)
+	if !ok {
+		common.Throw(rt, errors.New("failed to initialize tracing client, unable to load http module"))
+	}
+
+	// Get the http module's request function
+	httpModuleRequest := httpModuleObj.Get("request")
+
+	// Export the http module's request function goja.Callable as a Go function
+	var requestFunc HTTPRequestFunc
+	if err := rt.ExportTo(httpModuleRequest, &requestFunc); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, unable to export http.request method; reason: %w", err),
+		)
+	}
+
+	client := &Client{vu: vu, requestFunc: requestFunc}
+	if err := client.Configure(opts); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, invalid configuration; reason: %w", err),
+		)
+	}
+
+	return client
+}
+
+// Configure configures the tracing client with the given options.
+func (c *Client) Configure(opts options) error {
```

Comment:
Is it required to have it exported?

Reply:
It is intentional indeed so that users can reconfigure their client on the fly. It will especially prove useful at a later point to give the ability to change the sampling rate and baggage attached to requests 👍🏻 🙇🏻